### ISSUE TITLE
Cannot login with `bw login --apikey` : Missing expected CR after header value

### ISSUE THREAD

**Issue by DB-CL** [bitwarden/clients#4678]
### Steps To Reproduce

1. Open bash
2. Set the env `BW_CLIENTID` and `BW_CLIENTSECRET`
3. Set the server with `bw config server https://myserver`
4. Try to login with `bw login --apikey`

### Expected Result

The login should be sucessful

### Actual Result

The response is :
`request to https://myserver failed, reason: Parse Error: Missing expected CR after header value`

### Additional Context

Behind Nginx reverse proxy:
```
location /{ 
       proxy_pass http://localhost:9080;
       proxy_redirect off;
       proxy_set_header Host $http_host;
       proxy_set_header X-Real-IP $remote_addr;
       proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;       
}
```

 The CLI has been installed with snap

### Operating System

Linux

### Operating System Version

Ubuntu 22.04.1 LTS

### Shell

Zsh

### Build Version

2023.1.0

### Issue Tracking Info

- [x] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by joshtrichards** [comment]
@DB-CL I think this error is being spit out by the client, but is being triggered by something server-side (or an intermediary like at the RP) that isn't compliant.

**What's behind the Nginx reverse proxy? BW Unified? BW Standard? vaultwarden?**

**And, in any case, what version [it was unclear whether the version info you provided was for the CLI client or the server itself]?**

Also, did this work previously? If so, can you recall approximately when?

This looks like it could be related to a security fix made in a part of Node.js last year (llhttp) to stop being so liberal on the client and/or server-side (depending on which are using this code in a given situation) about non-compliant HTTP headers.
 
https://github.com/nodejs/llhttp/blob/2bf4d81d1e0d718a5d6c7732170176b81c52d612/src/llhttp/http.ts#L732

https://github.com/nodejs/llhttp/pull/161

P.S. It's also possible there is something failing (e.g. a weird poorly handled exception/error condition) on the server-side in a bad way that is causing the HTTP response to abort/be malformed I suppose. Can you check your server-side logs since you're self-hosted? Maybe even tail them while you try an `--apikey` connect?

**Comment by DB-CL** [comment]
> @DB-CL I think this error is being spit out by the client, but is being triggered by something server-side (or an intermediary like at the RP) that isn't compliant.

I agree, it also looks like it to me. My first thought was that the RP was not properly configured but I checked and I did not find any useful information about it.

> And, in any case, what version [it was unclear whether the version info you provided was for the CLI client or the server itself]?

It was for the bw cli but the server is also in 2023.1.0 according to the docker image that are running. I think this is BW standard (at the time I did the installation there was only one product)

> Also, did this work previously? If so, can you recall approximately when?

It is the first time I'm trying to use the cli client. The web client is working flawlessly for years.

> Can you check your server-side logs 

I ran a quick look and I did not find anything obvious. I checked the RP nginx and the BW docker nginx. I've tried a few files in the logs folder and found nothing. Do you have any pointers ?

**Comment by atjbramley** [comment]
Hi there,

I am unable to reproduce this issue, it has been escalated for further investigation. If you have more information that can help us, please add it below. 

Thanks!

**Comment by joshtrichards** [comment]
Hey @DB-CL Are you able to successfully complete a non-API interactive cli client successfully? - i.e. 

1. Start a new shell
2. Don't set anything BW related in the environment or pass any variables on the command-line
3.  `bw config server https://myserver`
4. `bw login`

You should get prompted for your email address then master password like a conventional login.

**Comment by DB-CL** [comment]
Hi, thank you both for your inputs.

I've made some progress 

>  Are you able to successfully complete a non-API interactive cli client successfully?

No, I have exactly the same error with your procedure

I tried to do the same directly on the bitwarden server, with http://localhost:9080 as a server and it worked ! So I really think this is the RP's fault, but my RP is just nginx with a (valid, I checked) SSL certificate and the code I shared in my first post. I will try to mess around with nginx tonight.
